raiden/network/proxies/token_network_registry.py

```diff
@@ -109,107 +139,296 @@ def add_token(
                 "the result of the precondition check is not precisely predictable."
             )
 
-        # check preconditions
+        if token_address == NULL_ADDRESS_BYTES:
+            raise InvalidTokenAddress("The call to register a token at 0x00..00 will fail.")
+
+        token_proxy = self.blockchain_service.token(token_address)
         try:
+            token_supply = token_proxy.total_supply(block_identifier=block_identifier)
```

Comment:
side note, can you try this out and see if the proxy will raise an exception instead of returning `""`? The RPC call does return `""`, and back in the day when we used the `JSONRPCClient` to do these requests that was the value we got. However, I believe that `web3` will actually check if the return type of the called function matches the value returned by the RPC call, and if the types don't match I believe it will raise an exception. In this case the `total_supply` ABI says the return type is an int, so I think that an exception will be raised here. You can easily check that by instantiating a `Token` proxy using the `TokenNetworkRegistry` address and calling the function.

Reply:
Created an issue for this: https://github.com/raiden-network/raiden/issues/4945